server/handle_sync_get_blob.go

```diff
@@ -47,15 +53,66 @@ func (s *Server) handleSyncGetBlob(e echo.Context) error {
 
 	buf := new(bytes.Buffer)
 
-	var parts []models.BlobPart
-	if err := s.db.Raw("SELECT * FROM blob_parts WHERE blob_id = ? ORDER BY idx", nil, blob.ID).Scan(&parts).Error; err != nil {
-		s.logger.Error("error getting blob parts", "error", err)
-		return helpers.ServerError(e, nil)
-	}
+	if blob.Storage == "sqlite" {
+		var parts []models.BlobPart
+		if err := s.db.Raw("SELECT * FROM blob_parts WHERE blob_id = ? ORDER BY idx", nil, blob.ID).Scan(&parts).Error; err != nil {
+			s.logger.Error("error getting blob parts", "error", err)
+			return helpers.ServerError(e, nil)
+		}
+
+		// TODO: we can just stream this, don't need to make a buffer
+		for _, p := range parts {
+			buf.Write(p.Data)
+		}
+	} else if blob.Storage == "s3" && s.s3Config != nil && s.s3Config.BlobstoreEnabled {
```

Comment:
small nit, might make sense to do something like this maybe? since technically it wouldn't be unknown storage, just missing config

```suggestion
	} else if blob.Storage == "s3" {
		if s.s3Config == nil || !s.s3Config.BlobstoreEnabled {
			s.logger.Error("unknown storage", "storage", blob.Storage)
			return helpers.ServerError(e, nil)
		}
```